docs/src/v14_migration_guide.md

```diff
@@ -0,0 +1,193 @@
+
+## New interface for accessing problem/integrator/solution parameter (and species) value
+Previously, it was possible to directly index problems to query them for their parameter values. e.g.
+```@example v14_migration_4
+using Catalyst
+rn = @reaction_network begin
+ (p,d), 0 <--> X
+end
+u0 = [:X => 1.0]
+ps = [:p => 1.0, :d => 0.2]
+oprob = ODEProblem(rn, u0, (0.0, 1.0), ps)
+nothing # hide
+```
+```julia
+oprob[:p]
+```
+This is *no longer supported*. When you wish to query a problem (or integrator or solution) for a parameter value (or to update a parameter value), you must append `.ps` to the problem variable name:
+```@example v14_migration_4
+oprob.ps[:p]
+```
+
+Furthermore, a few new functions (`getp`, `getu`, `setp`, `setu`) have been introduced. These can improve performance when querying a structure for a value multiple times (especially for very large models). These are described in more detail [here](@ref simulation_structure_interfacing_functions).
+
+For more details on how to query various structures for parameter and species values, please read [this documentation page](@ref simulation_structure_interfacing).
+
+## Other notes
+Finally, here are some additional, minor, notes regarding the new update.
+
+#### Modification of problems with conservation laws broken
+While it is possible to update e.g. `ODEProblem`s using the [`remake`](@ref simulation_structure_interfacing_problems_remake) function, this is currently not possible if the `remove_conserved = true` option was used. E.g. while
+```@example v14_migration_5
+using Catalyst, OrdinaryDiffEq
+rn = @reaction_network begin
+ (k1,k2), X1 <--> X2
+end
+u0 = [:X1 => 1.0, :X2 => 2.0]
+ps = [:k1 => 0.5, :k2 => 3.0]
+oprob = ODEProblem(rn, u0, (0.0, 10.0), ps; remove_conserved = true)
+sol(oprob)
+# hide
+```
+is perfectly fine, attempting to then modify `oprob` (in any manner!) is not possible:
+```@example v14_migration_5
+oprob_remade = remake(oprob; u0 = [:X1 => 5.0]) # NEVER do this.
+sol(oprob)
+# hide
+```
+This might generate a silent error, where the remade problem is different from the intended one. 
+
+This bug was likely present on earlier versions as well, but was only recently discovered. While we hope it will be fixed soon, the bug is in ModelingToolkit, and will not be fixed until its maintainers find the time to do so.
+
+#### Depending on parameter order is even more dangerous than before
+In early versions of Catalyst, parameters and species were provided as vectors (e.g. `[1.0, 2.0]`) rather than maps (e.g. `[p => 1.0, d => 2.0]`). While we have already *strongly* recommended users to use the map form (or they might produce unintended results), the vector form is still (technically). Due to recent internal ModelingToolkit updates, the risk of unexpected behaviour when providing parameter values as vectors is even larger than before.
```

Comment:
```suggestion
In early versions of Catalyst, parameters and species were provided as vectors (e.g. `[1.0, 2.0]`) rather than maps (e.g. `[p => 1.0, d => 2.0]`). While we previously *strongly* recommended users to use the map form (or they might produce unintended results), the vector form was still supported (technically). Due to recent internal ModelingToolkit updates, the purely numeric form is no longer supported and should never be used -- it will potentially lead to incorrect values for parameters and/or initial conditions. Note that if `rn` is a complete `ReactionSystem` you can now specify such mappings via `[rn.p => 1.0, rn.d => 2.0]`.
```